### ISSUE TITLE
Startup failed: std::runtime_error (Cannot replace_address 10.4.0.21 because it doesn't exist in gossip)

### ISSUE THREAD

**Issue by juliayakovlev** [scylladb/scylladb#11960]
## Installation details

Kernel Version: 5.15.0-1022-aws
Scylla version (or git commit hash): `2022.2.0~rc4-20221106.f5714e0db12f` with build-id `f4a927b8a00fbcd8d48640835192aeaa7968b1f2`

Cluster size: 4 nodes (i3.4xlarge)

Scylla Nodes used in this run:
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-9 (52.212.112.192 | 10.4.0.51) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8 (3.251.79.14 | 10.4.2.113) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-7 (34.245.56.196 | 10.4.2.70) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 (18.202.219.69 | 10.4.1.114) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-5 (34.246.250.177 | 10.4.0.93) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-4 (3.252.162.201 | 10.4.0.64) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-3 (34.251.128.103 | 10.4.0.21) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-2 (52.19.215.161 | 10.4.3.148) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-11 (34.246.193.25 | 10.4.0.111) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-10 (34.244.150.185 | 10.4.0.137) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-c159e20e-1 (54.74.4.53 | 10.4.2.34) (shards: 14)

OS / Image: `ami-00253925d2028dcef` (aws: eu-west-1)

Test: `ics-longevity-1tb-7days-test`
Test id: `c159e20e-0347-4563-9ad5-5113f5867513`
Test name: `enterprise-2022.2/SCT_Enterprise_Features/ICS/ics-longevity-1tb-7days-test`
Test config file(s):

- [longevity-1TB-5days-authorization-and-tls-ssl.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/519e3909374cd87064c140c4badb2acc05805682/test-cases/longevity/longevity-1TB-5days-authorization-and-tls-ssl.yaml)

## Issue description

Node `longevity-tls-1tb-7d-2022-2-db-node-c159e20e-3 [34.251.128.103 | 10.4.0.21]` was terminated and replaced with new added `longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8`.

But startup of new node failed with `std::runtime_error (Cannot replace_address 10.4.0.21 because it doesn't exist in gossip)`:

```
Nov 12 11:08:50 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8 scylla[7831]:  [shard  0] gossip - Gossip shadow round finisehd with nodes_talked={10.4.0.93, 10.4.2.34}
Nov 12 11:08:50 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8 scylla[7831]:  [shard  0] gossip - Feature check passed. Local node 10.4.2.113 features = {AGGREGATE_STORAGE_OPTIONS, ALTERNATOR_STREAMS, ALTERNATO>
Nov 12 11:08:50 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8 scylla[7831]:  [shard  0] init - Shutting down group 0 service
Nov 12 11:08:50 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8 scylla[7831]:  [shard  0] init - Shutting down group 0 service was successful

Nov 12 11:08:50 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8 scylla[7831]:  [shard  0] init - Shutting down sighup was successful
Nov 12 11:08:50 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8 scylla[7831]:  [shard  0] init - Startup failed: std::runtime_error (Cannot replace_address 10.4.0.21 because it doesn't exist in gossip)
```

No coredumps or other error found on the node.

Nemesis:
```
< t:2022-11-12 04:45:05,536 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2022-11-12 04:45:05.535: (DisruptionEvent Severity.NORMAL) period_type=begin event_id=b79746c3-9493-4619-9b82-c4f43f8a
49eb: nemesis_name=TerminateAndReplaceNode target_node=Node longevity-tls-1tb-7d-2022-2-db-node-c159e20e-3 [34.251.128.103 | 10.4.0.21] (seed: False)
```

- Restore Monitor Stack command: `$ hydra investigate show-monitor c159e20e-0347-4563-9ad5-5113f5867513`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=c159e20e-0347-4563-9ad5-5113f5867513)
- Show all stored logs command: `$ hydra investigate show-logs c159e20e-0347-4563-9ad5-5113f5867513`

## Logs:
Test is still running and node `longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8 (3.251.79.14 | 10.4.2.113)` exists

[Jenkins job URL](https://jenkins.scylladb.com/job/enterprise-2022.2/job/SCT_Enterprise_Features/job/ICS/job/ics-longevity-1tb-7days-test/5/)

**Comment by DoronArazii** [comment]
@asisas please have a look

**Comment by roydahan** [comment]
@juliayakovlev please add the information of “node tool status” before the nemesis and if we have, also during the nemesis.

**Comment by fgelcer** [comment]
> I found the reason. After node termination, new node `longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 [18.202.219.69 | 10.4.1.114]` was added and initialized. In the test we are waiting 21600 seconds (6 hours) for the new node is ready. In this case the node initialization took 6h2m.
> 
> ```
> < t:2022-11-12 04:52:02,972 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2022-11-12 04:52:02.967: (InfoEvent Severity.NORMAL) period_type=not-set event_id=02d643e8-e3c1-49ce-b045-43c0828b348a: message=StartEvent - Adding new node to cluster
> < t:2022-11-12 04:52:34,897 f:cluster.py      l:339  c:sdcm.cluster_aws     p:DEBUG > Node longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 [18.202.219.69 | 10.4.1.114] (seed: False): SSH access -> 'ssh -i ~/.ssh/scylla-qa-ec2 scyllaadm@10.4.1.114'
> 
> Nov 12 04:54:54 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]: Scylla version 2022.2.0~rc4-0.20221106.f5714e0db12f with build-id f4a927b8a00fbcd8d48640835192aeaa7968b1f2 starting ...
> 
> < t:2022-11-12 10:56:30,075 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG > '_add_and_init_new_cluster_node': failed with '[Node longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 [18.202.219.69 | 10.4.1.114] (seed: False)] NodeSetupFailed: RetryError[Wait for: db_up: timeout - 21600 seconds - expired]
> 
> Nov 12 10:56:51 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] init - Scylla version 2022.2.0~rc4-0.20221106.f5714e0db12f initialization completed.
> ```
> 
> So waiting was interrupted and [suppose that new node setup failed] try to add another node. We try to add new node 3 times, so `longevity-tls-1tb-7d-2022-2-db-node-c159e20e-7` and node `longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8` failed with startup error described above ("Cannot replace_address 10.4.0.21 because it doesn't exist in gossip").
> 
> @DoronArazii I think this is not real issue and can be closed. My appologizes for a mess

the question is why replace node is taking 6+ hours?

**Comment by mykaul** [comment]
@juliayakovlev good finding and an important one. Let's see if we can figure out why it took so long to complete the process of node replacement.

**Comment by asias** [comment]
Using sct commit e04dd826d657ca30089db0bf1aab02a07ad4c851, it failed to restore the monitor stack. 

```
$ hydra investigate show-monitor c159e20e-0347-4563-9ad5-5113f5867513
There is scylladb/hydra:v1.21 in local cache, using it.
Obtaining QA SSH keys...
QA SSH keys obtained.
Going to run './sct.py  investigate show-monitor c159e20e-0347-4563-9ad5-5113f5867513'...
Obtaining QA SSH keys...
QA SSH keys obtained.
/home/asias/.local/lib/python3.10/site-packages/requests/__init__.py:102: RequestsDependencyWarning: urllib3 (1.26.12) or chardet (5.0.0)/charset_normalizer (2.1.1) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({})/charset_normalizer ({}) doesn't match a supported "
install-bash-completion current path: /home/asias/src/cloudius-systems/scylla-cluster-tests
New directory created: /home/asias/sct-results/20221114-013303-059111-investigate-show-monitor
Search monitoring stack archive files for test id c159e20e-0347-4563-9ad5-5113f5867513 and restoring...
Checking that docker is available...
Docker is available
No any archive for monitoring stack
Errors were found when restoring Scylla monitoring stack
Killing Grafana service
Killing Prometheus service
Killing Alert service
```

**Comment by juliayakovlev** [comment]
> Using sct commit e04dd826d657ca30089db0bf1aab02a07ad4c851, it failed to restore the monitor stack.
> 
> ```
> $ hydra investigate show-monitor c159e20e-0347-4563-9ad5-5113f5867513
> There is scylladb/hydra:v1.21 in local cache, using it.
> Obtaining QA SSH keys...
> QA SSH keys obtained.
> Going to run './sct.py  investigate show-monitor c159e20e-0347-4563-9ad5-5113f5867513'...
> Obtaining QA SSH keys...
> QA SSH keys obtained.
> /home/asias/.local/lib/python3.10/site-packages/requests/__init__.py:102: RequestsDependencyWarning: urllib3 (1.26.12) or chardet (5.0.0)/charset_normalizer (2.1.1) doesn't match a supported version!
>   warnings.warn("urllib3 ({}) or chardet ({})/charset_normalizer ({}) doesn't match a supported "
> install-bash-completion current path: /home/asias/src/cloudius-systems/scylla-cluster-tests
> New directory created: /home/asias/sct-results/20221114-013303-059111-investigate-show-monitor
> Search monitoring stack archive files for test id c159e20e-0347-4563-9ad5-5113f5867513 and restoring...
> Checking that docker is available...
> Docker is available
> No any archive for monitoring stack
> Errors were found when restoring Scylla monitoring stack
> Killing Grafana service
> Killing Prometheus service
> Killing Alert service
> ```

Test is running. 
You can see the monitor: http://34.240.10.154:3000/

**Comment by juliayakovlev** [comment]
> > I found the reason. After node termination, new node `longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 [18.202.219.69 | 10.4.1.114]` was added and initialized. In the test we are waiting 21600 seconds (6 hours) for the new node is ready. In this case the node initialization took 6h2m.
> > ```
> > < t:2022-11-12 04:52:02,972 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2022-11-12 04:52:02.967: (InfoEvent Severity.NORMAL) period_type=not-set event_id=02d643e8-e3c1-49ce-b045-43c0828b348a: message=StartEvent - Adding new node to cluster
> > < t:2022-11-12 04:52:34,897 f:cluster.py      l:339  c:sdcm.cluster_aws     p:DEBUG > Node longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 [18.202.219.69 | 10.4.1.114] (seed: False): SSH access -> 'ssh -i ~/.ssh/scylla-qa-ec2 scyllaadm@10.4.1.114'
> > 
> > Nov 12 04:54:54 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]: Scylla version 2022.2.0~rc4-0.20221106.f5714e0db12f with build-id f4a927b8a00fbcd8d48640835192aeaa7968b1f2 starting ...
> > 
> > < t:2022-11-12 10:56:30,075 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG > '_add_and_init_new_cluster_node': failed with '[Node longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 [18.202.219.69 | 10.4.1.114] (seed: False)] NodeSetupFailed: RetryError[Wait for: db_up: timeout - 21600 seconds - expired]
> > 
> > Nov 12 10:56:51 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] init - Scylla version 2022.2.0~rc4-0.20221106.f5714e0db12f initialization completed.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > So waiting was interrupted and [suppose that new node setup failed] try to add another node. We try to add new node 3 times, so `longevity-tls-1tb-7d-2022-2-db-node-c159e20e-7` and node `longevity-tls-1tb-7d-2022-2-db-node-c159e20e-8` failed with startup error described above ("Cannot replace_address 10.4.0.21 because it doesn't exist in gossip").
> > @DoronArazii I think this is not real issue and can be closed. My appologizes for a mess
> 
> the question is why replace node is taking 6+ hours?

Yes, but it's another issue, not as I described

**Comment by asias** [comment]
> > Using sct commit e04dd826d657ca30089db0bf1aab02a07ad4c851, it failed to restore the monitor stack.
> > ```
> > $ hydra investigate show-monitor c159e20e-0347-4563-9ad5-5113f5867513
> > There is scylladb/hydra:v1.21 in local cache, using it.
> > Obtaining QA SSH keys...
> > QA SSH keys obtained.
> > Going to run './sct.py  investigate show-monitor c159e20e-0347-4563-9ad5-5113f5867513'...
> > Obtaining QA SSH keys...
> > QA SSH keys obtained.
> > /home/asias/.local/lib/python3.10/site-packages/requests/__init__.py:102: RequestsDependencyWarning: urllib3 (1.26.12) or chardet (5.0.0)/charset_normalizer (2.1.1) doesn't match a supported version!
> >   warnings.warn("urllib3 ({}) or chardet ({})/charset_normalizer ({}) doesn't match a supported "
> > install-bash-completion current path: /home/asias/src/cloudius-systems/scylla-cluster-tests
> > New directory created: /home/asias/sct-results/20221114-013303-059111-investigate-show-monitor
> > Search monitoring stack archive files for test id c159e20e-0347-4563-9ad5-5113f5867513 and restoring...
> > Checking that docker is available...
> > Docker is available
> > No any archive for monitoring stack
> > Errors were found when restoring Scylla monitoring stack
> > Killing Grafana service
> > Killing Prometheus service
> > Killing Alert service
> > ```
> 
> Test is running. You can see the monitor: http://34.240.10.154:3000/

OK. When will this test finish? Can you extract the logs from node6 before we have the logs links?

**Comment by juliayakovlev** [comment]
You can connect to the node as `ssh -i ~/.ssh/scylla-qa-ec2 centos@18.202.219.69` and investigate it there. The bootstrap took there 6h:
```
Nov 12 04:56:47 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] storage_service - Starting to bootstrap...
Nov 12 10:56:37 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] storage_service - Bootstrap completed!
```

**Comment by juliayakovlev** [comment]
> > > Using sct commit e04dd826d657ca30089db0bf1aab02a07ad4c851, it failed to restore the monitor stack.
> > > ```
> > > $ hydra investigate show-monitor c159e20e-0347-4563-9ad5-5113f5867513
> > > There is scylladb/hydra:v1.21 in local cache, using it.
> > > Obtaining QA SSH keys...
> > > QA SSH keys obtained.
> > > Going to run './sct.py  investigate show-monitor c159e20e-0347-4563-9ad5-5113f5867513'...
> > > Obtaining QA SSH keys...
> > > QA SSH keys obtained.
> > > /home/asias/.local/lib/python3.10/site-packages/requests/__init__.py:102: RequestsDependencyWarning: urllib3 (1.26.12) or chardet (5.0.0)/charset_normalizer (2.1.1) doesn't match a supported version!
> > >   warnings.warn("urllib3 ({}) or chardet ({})/charset_normalizer ({}) doesn't match a supported "
> > > install-bash-completion current path: /home/asias/src/cloudius-systems/scylla-cluster-tests
> > > New directory created: /home/asias/sct-results/20221114-013303-059111-investigate-show-monitor
> > > Search monitoring stack archive files for test id c159e20e-0347-4563-9ad5-5113f5867513 and restoring...
> > > Checking that docker is available...
> > > Docker is available
> > > No any archive for monitoring stack
> > > Errors were found when restoring Scylla monitoring stack
> > > Killing Grafana service
> > > Killing Prometheus service
> > > Killing Alert service
> > > ```
> > 
> > 
> > Test is running. You can see the monitor: http://34.240.10.154:3000/
> 
> OK. When will this test finish? Can you extract the logs from node6 before we have the logs links?

You can connect to the node as `ssh -i ~/.ssh/scylla-qa-ec2 centos@18.202.219.69` and investigate it there. The bootstrap took there 6h:
```
Nov 12 04:56:47 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] storage_service - Starting to bootstrap...
Nov 12 10:56:37 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] storage_service - Bootstrap completed!
```

**Comment by asias** [comment]
node6 was replacing from ~05:00  Nov 12 to 11:00 Nov 12. The dominant keyspace for the whole replace ops is keyspace1.

Keyspace1 started at 05:00, finished at 08:30. In the middle, off strategy kicked in at 06:20 which used much more disk bandwidth for compaction (from100MB/s to 1.5GB/s).

My conclusion is that off-strategy makes the replace ops took much longer.  On one hand, it makes the replace operation longer. On the other hand, when the replace is done, the sstables are in sharp. 

```
scyllaadm@longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6:~$ cat 6 |grep replace_with_repair:
Nov 12 04:56:48 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: started with keyspaces={system_traces, keyspace1, system_auth, keyspace_deflate, mview, system_distributed, system_distributed_everywhere, keyspace_lz4, keyspace_snappy}, source_dc=eu-west, nr_ranges_total=15290, ignore_nodes=[]
Nov 12 04:56:48 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: started with keyspace=system_traces, source_dc=eu-west, nr_ranges=2625, ignore_nodes=[]
Nov 12 04:56:52 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: finished with keyspace=system_traces, source_dc=eu-west, nr_ranges=525
Nov 12 04:56:52 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: started with keyspace=keyspace1, source_dc=eu-west, nr_ranges=776, ignore_nodes=[]
Nov 12 08:30:20 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: finished with keyspace=keyspace1, source_dc=eu-west, nr_ranges=776
Nov 12 08:30:20 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: started with keyspace=system_auth, source_dc=eu-west, nr_ranges=3104, ignore_nodes=[]
Nov 12 08:30:33 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: finished with keyspace=system_auth, source_dc=eu-west, nr_ranges=776
Nov 12 08:30:33 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: started with keyspace=keyspace_deflate, source_dc=eu-west, nr_ranges=776, ignore_nodes=[]
Nov 12 09:35:55 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: finished with keyspace=keyspace_deflate, source_dc=eu-west, nr_ranges=776
Nov 12 09:35:55 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: started with keyspace=mview, source_dc=eu-west, nr_ranges=2328, ignore_nodes=[]
Nov 12 09:42:21 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: finished with keyspace=mview, source_dc=eu-west, nr_ranges=776
Nov 12 09:42:21 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: started with keyspace=system_distributed, source_dc=eu-west, nr_ranges=3104, ignore_nodes=[]
Nov 12 09:42:39 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: finished with keyspace=system_distributed, source_dc=eu-west, nr_ranges=776
Nov 12 09:42:39 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: started with keyspace=system_distributed_everywhere, source_dc=eu-west, nr_ranges=1025, ignore_nodes=[]
Nov 12 09:42:46 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: finished with keyspace=system_distributed_everywhere, source_dc=eu-west, nr_ranges=1025
Nov 12 09:42:46 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: started with keyspace=keyspace_lz4, source_dc=eu-west, nr_ranges=776, ignore_nodes=[]
Nov 12 10:17:50 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: finished with keyspace=keyspace_lz4, source_dc=eu-west, nr_ranges=776
Nov 12 10:17:50 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: started with keyspace=keyspace_snappy, source_dc=eu-west, nr_ranges=776, ignore_nodes=[]
Nov 12 10:56:36 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: finished with keyspace=keyspace_snappy, source_dc=eu-west, nr_ranges=776
Nov 12 10:56:36 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] repair - replace_with_repair: finished with keyspaces={system_traces, keyspace1, system_auth, keyspace_deflate, mview, system_distributed, system_distributed_everywhere, keyspace_lz4, keyspace_snappy}, source_dc=eu-west
```

```
Nov 12 06:21:18 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard 10] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 704 candidates were found
Nov 12 06:23:36 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  8] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 692 candidates were found
Nov 12 06:24:39 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  6] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 696 candidates were found
Nov 12 06:25:48 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  7] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 692 candidates were found
Nov 12 06:26:55 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard 11] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 698 candidates were found
Nov 12 06:27:58 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  9] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 694 candidates were found
Nov 12 06:28:17 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  1] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 701 candidates were found
Nov 12 06:30:07 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  0] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 708 candidates were found
Nov 12 06:42:10 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  2] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 697 candidates were found
Nov 12 06:45:34 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  4] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 701 candidates were found
Nov 12 06:45:36 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  3] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 695 candidates were found
Nov 12 06:46:03 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard 12] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 702 candidates were found
Nov 12 06:46:52 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard  5] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 692 candidates were found
Nov 12 08:30:20 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard 13] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 704 candidates were found
```

![image](https://user-images.githubusercontent.com/320076/201624337-8ff60302-6a69-4fd8-88f1-b6db74c08c6e.png)

**Comment by juliayakovlev** [comment]
I updated description

**Comment by avikivity** [comment]
It's wrong that offstrategy compaction started before streaming/repair was completed. It means offstrategy competes with streaming/repair, and then it gets more work as more sstables are added and it needs to redo some work.

**Comment by roydahan** [comment]
@mykaul this issue by itself doesn't mean anything.
The time it takes to replace a node depends on the dataset size and the load during replacement.
AFAIK, there is no hard requirement we can use in order to determine pass vs failed, so IMO the work should start from defining the goal.

**Comment by roydahan** [comment]
OTOH, what we can do is report some statistics on how long it use to take previously...

**Comment by avikivity** [comment]
> As I described in description, after firmware update almost 1 hour nothing happens and just after 1 hour started reshaping:
> 
> ```
> Nov 12 05:23:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 fwupdmgr[10075]: Successfully downloaded new metadata: 0 local devices supported
> ###### nothing happens here #############
> Nov 12 06:21:18 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard 10] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 704 candidates were found
> ```

Anything in the metrics in that hour?

**Comment by juliayakovlev** [comment]
> > As I described in description, after firmware update almost 1 hour nothing happens and just after 1 hour started reshaping:
> > ```
> > Nov 12 05:23:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 fwupdmgr[10075]: Successfully downloaded new metadata: 0 local devices supported
> > ###### nothing happens here #############
> > Nov 12 06:21:18 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 scylla[7825]:  [shard 10] compaction_manager - Starting off-strategy compaction for keyspace1.standard1, 704 candidates were found
> > ```
> 
> Anything in the metrics in that hour?

What should I search?
This is a monitor for the new node during the hour from `12 05:23:00` to `12 06:21:18`

http://34.240.10.154:3000/d/detailed-master/detailed?orgId=1&from=1668230432952&to=1668234699201&var-by=instance&var-cluster=&var-dc=All&var-node=10.4.1.114&var-shard=All&var-scheduling_group=All&var-func=sum&var-scylla_version=master&var-monitoring_version=4.0.2

**Comment by avikivity** [comment]
Looks like there was a steady write load:

![image](https://user-images.githubusercontent.com/1017210/201924162-5e24a1d1-e906-4133-bd99-eebb0487d149.png)

So, why do you say nothing was happening? What should have happened? Did the test try to make something happen?

**Comment by avikivity** [comment]
Strangely, very few compactions, and little compaction CPU time. Maybe this is an overwrite load?

![image](https://user-images.githubusercontent.com/1017210/201924669-502a046c-ec43-472b-ac7f-8d4bddf5aceb.png)

**Comment by juliayakovlev** [comment]
> Looks like there was a steady write load:
> 
> ![image](https://user-images.githubusercontent.com/1017210/201924162-5e24a1d1-e906-4133-bd99-eebb0487d149.png)
> 
> So, why do you say nothing was happening? What should have happened? Did the test try to make something happen?

I did not find in the log that runing reshape/compaction or something else. Reshaping was started later

**Comment by avikivity** [comment]
Yes, but why did you expect anything in the log?

**Comment by juliayakovlev** [comment]
> Yes, but why did you expect anything in the log?

I expected to see that bootstrap advances

**Comment by mykaul** [comment]
> fwupdmgr

why are we running fwupdmgr on the cloud? What do we expect will get updated? @yaronkaikov  ?

**Comment by dorlaor** [comment]
My 2 cents:

- Requirement for replace speed:
  We need to get as close as possible to the hardware speed from the slowest
  component. For 1kb objects, we should be able to run at the minimum disk or net
  speed. There is no reason for such a node to do less than 1GB/s which will translate
  into 1000 seconds a 1TB. If we're twice as slow, it should be 2000 sec - 40 minutes 
  and not 6 hours

- Firmware upgrade - let's open a separate issue 
- Off strategy - doesn't replace/ RBNO controls it?

**Comment by yaronkaikov** [comment]
> fwupdmgr

@mykaul Opened https://github.com/scylladb/scylla-machine-image/issues/408 for the cleanup of the services we don't really need in our images

**Comment by asias** [comment]
> It's wrong that offstrategy compaction started before streaming/repair was completed. 

There is a reason why we trigger off-strategy compaction:

The issue: Too many open files during terminate and replace node (scale test) #10462
The fix: e65b3e

> It means offstrategy competes with streaming/repair, and then it gets more work as more sstables are added and it needs to redo some work.

We trigger off-strategy for a specific table only after all the sstables for that table is received. We redo no work. 

The problem we saw here is that offstrategy under streaming class is too aggressive. I think we'd put offstrategy in a new scheduling group with lower priority than streaming class (200 shares), e.g., make it 100 or 150 shares. 

@raphaelsc FYI.

**Comment by dorlaor** [comment]
On Wed, Nov 16, 2022 at 3:55 AM Asias He ***@***.***> wrote:

> It's wrong that offstrategy compaction started before streaming/repair was
> completed.
>
> There is a reason why we trigger off-strategy compaction:
>
> The issue: Too many open files during terminate and replace node (scale
> test) #10462 <https://github.com/scylladb/scylladb/issues/10462>
> The fix: e65b3e
>
> It means offstrategy competes with streaming/repair, and then it gets more
> work as more sstables are added and it needs to redo some work.
>
> We trigger off-strategy for a specific table only after all the sstables
> for that table is received. We redo no work.
>
> The problem we saw here is that offstrategy under streaming class is too
> aggressive. I think we'd put offstrategy in a new scheduling group with
> lower priority than streaming class (200 shares), e.g., make it 100 or 150
> shares.
>
> @raphaelsc <https://github.com/raphaelsc> FYI.
>

An alternative would be to check the number of files and compare it to a
desired criteria
and change priorities based on this. If we add too many priority
dependencies we'll run into
another issue. An alternative option is for repair to check the current
state before it begins

**Comment by asias** [comment]
> An alternative would be to check the number of files and compare it to a desired criteria and change priorities based on this. 

Do you mean changing priority of off-strategy compaction dynamically? It currently shares the same priority of streaming, we can change it until we have a separate schedule group for it. 

> If we add too many priority dependencies we'll run into another issue. An alternative option is for repair to check the current state before it begins

It is hard to define what is the proper state where repair for the replace should stop. Also,  stopping reapir does not fix the problem that offstrategy compaction is too aggressive. It makes it is even more aggressive because no streaming is going to complete with it.

**Comment by dorlaor** [comment]
On Wed, Nov 16, 2022 at 10:09 AM Asias He ***@***.***> wrote:

> An alternative would be to check the number of files and compare it to a
> desired criteria and change priorities based on this.
>
> Do you mean changing priority of off-strategy compaction dynamically? It
> currently shares the same priority of streaming, we can change it until we
> have a separate schedule group for it.
>
> If we add too many priority dependencies we'll run into another issue. An
> alternative option is for repair to check the current state before it begins
>
> It is hard to define what is the proper state where repair for the replace
> should stop. Also, stopping reapir does not fix the problem that
> offstrategy compaction is too aggressive. It makes it is even more
> aggressive because no streaming is going to complete with it.
>

We should simplify our scenarios and not try to handle all of them in the
best way, since we wouldn't be successful
or the code will be buggy. It's best not to run off strategy in parallel to
repair. Either the current state is good for repair
to pause compaction (or just the off strategy) and run repair only, or the
opposite where we must finish compaction
and only later run repair.

We don't need to optimize a case where repair was terminated in the middle
and restarted later after decommission.
Best is to have these mechanisms:
  - Comaction controller always check for urgent compaction need state
     - Defined by too many files or too many reads per data access or lack
of space
     - In such a case, it halts repair and other efforts and runs
emergency compaction
     - This has higher priority than all admin workloads
  - When repair runs, it automatically stops compaction

This way there aren't too many different priorities

**Comment by asias** [comment]
> We should simplify our scenarios and not try to handle all of them in the
best way, since we wouldn't be successful
or the code will be buggy. It's best not to run off strategy in parallel to
repair. Either the current state is good for repair
to pause compaction (or just the off strategy) and run repair only, or the
opposite where we must finish compaction
and only later run repair.

Dor, I am talking about the simplest scenario, just plain replace operation, nothing more. The thing you might have missed is that off strategy is per table basis. It is not per operation. The idea behind offstrategy is that we should only do compaction for the given table after all data for that table has arrived to avoid compacting again and again. This does not mean we have to wait for the whole node operation to finish (all keyspaces and all tables have finished), only then we can run offstrategy compaction.

> We don't need to optimize a case where repair was terminated in the middle
and restarted later after decommission.

Nothing is intended to optimize this. I am not sure why it is relevant to the issue here. 

> Best is to have these mechanisms:
>   - Comaction controller always check for urgent compaction need state
     - Defined by too many files or too many reads per data access or lack
of space
     - In such a case, it halts repair and other efforts and runs
emergency compaction
     - This has higher priority than all admin workloads
>   - When repair runs, it automatically stops compaction

The repair will run for the whole replace operation. The second bullet above will make us never run off-strategy compaction which means we could accumulate too many sstable files and make the node explode, like the issue https://github.com/scylladb/scylladb/issues/10462. Note #10462 has nothing to do with decommission. It is just a plain replace operation. It is not a rare and weird operation we try to optimize. 

> This way there aren't too many different priorities

**Comment by raphaelsc** [comment]
new node: longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6 [18.202.219.69 | 10.4.1.114]

1) bootstrap starts at 04:56:47
2) hangs for ~1 hour 05:23:00 - 06:21:18 due to firmware update
3) bootstrap ends at 10:56:37

total time: ~6h
actual bootstrap time: ~5h - [04:56:46 - 05:23] [06:21:18, 10:56:37]

1TB / 3600 / 5 = 59652323 = 59652323 / sec = 56M / sec

write_bandwidth: 1617551304 = 1542.61 MB

WA of ~27 ?

something is very wrong

**Comment by raphaelsc** [comment]
we're reshaping new streamed data in multiple rounds for the same data:
```
2022-11-12T06:33:11+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaped 704 sstables to [/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9894-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9908-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9922-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9936-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9950-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9964-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9978-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9992-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10006-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10020-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10034-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10048-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10062-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10076-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10090-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10104-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10118-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10132-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10146-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10160-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10174-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10188-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10202-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10216-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10230-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10244-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10258-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10272-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10286-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10300-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10314-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10328-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10342-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10356-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10370-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10384-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10398-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10412-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10426-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10440-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10454-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10468-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10482-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10496-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10510-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10524-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10538-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10552-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10566-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10580-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10594-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10608-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10622-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10636-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10650-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10664-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10678-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10692-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10706-big-Data.db:level=0,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10720-big-Data.db:level=0]. 64GB to 63GB (~99% of original) in 712585ms = 89MB/s. ~58599680 total partitions merged to 58555974.
2022-11-12T06:33:11+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaping [/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9894-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9908-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9922-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9936-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9950-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9964-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9978-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-9992-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10006-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10020-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10034-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10048-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10062-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10076-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10090-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10104-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10118-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10132-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10146-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10160-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10174-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10188-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10202-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10216-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10230-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10244-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10258-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10272-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10286-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10300-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10314-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10328-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10342-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10356-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10370-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10384-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10398-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10412-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10426-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10440-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10454-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10468-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10482-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10496-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10510-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10524-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10538-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10552-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10566-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10580-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10594-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10608-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10622-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10636-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10650-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10664-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10678-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10692-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10706-big-Data.db:level=0:origin=reshape,/var/lib/scylla/data/keyspace1/standard1-0d4a98f0611a11edafd7e19ac39a57d7/me-10720-big-Data.db:level=0:origin=reshape]
```

this shouldn't have happened, as input is initially disjoint. the first round happens on 704 sstables, meaning ICS detected it to be disjoint. we're doing useless work thereafter it seems...

**Comment by asias** [comment]
@raphaelsc  Are you saying the Reshaped and reshaping log with the same file list below means we were doing the same reshap on the same sstables TWICE? Under the same uuid 38078290-6252-11ed-b5f3-6775352712a0, the  initial 704 sstables from repair on shard 0 was reshaped, then we have multiple 60 sstables were merged again, what was going on here?

```
2022-11-12T06:33:11+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaped 704 sstables to 
2022-11-12T06:33:11+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaping [/var/lib/scyll
2022-11-12T07:02:09+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaped 60 sstables to [
2022-11-12T07:02:09+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaping [/var/lib/scyll
2022-11-12T07:20:12+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaped 60 sstables to [
2022-11-12T07:20:13+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaping [/var/lib/scyll
2022-11-12T07:37:27+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaped 60 sstables to [
2022-11-12T07:37:27+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaping [/var/lib/scyll
2022-11-12T07:54:45+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaped 60 sstables to [
2022-11-12T07:54:46+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaping [/var/lib/scyll
2022-11-12T08:12:01+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaped 60 sstables to [
2022-11-12T08:12:02+00:00 longevity-tls-1tb-7d-2022-2-db-node-c159e20e-6     !INFO | scylla[7825]:  [shard 10] compaction - [Reshape keyspace1.standard1 38078290-6252-11ed-b5f3-6775352712a0] Reshaping [/var/lib/scyll
```

**Comment by roydahan** [comment]
@mykaul / @avikivity this issue was pushed to 5.2, but if it's ICS specific, you may want to reconsider it to the 2022.2 release.
